### ISSUE TITLE
Allow multiple whiskers

### ISSUE THREAD

**Issue by paulsonnentag** [janpaul123/paperprograms#49]
I wanted to experiment with multiple whiskers but creating the second whisker failed with the error:

```
Error[program:0:0]: "Cannot read property 'getContext' of null"
```

The problem seems to be that `paper.get('supporterCanvas')` only returns a reference to the canvas on the first call. I assume the reason for this is that you can call `transferControlToOffscreen` only once. Instead of just returning null I think it would be nicer if paper.get('supporterCanvas') would cache the reference to the canvas on the worker side and return the cached reference instead. I think this would be generally useful independent of the whisker issue. To improve handling multipe whiskers I would like to change the current implementation so only a single setInterval is created by `paper.whenPointsAt()`.

**Comment by janpaul123** [comment]
Did this get fixed by #51 or not yet?

**Comment by paulsonnentag** [comment]
No this is a separate issue. The problem is the following line, where the whisker canvas is requested:

https://github.com/janpaul123/paperprograms/blob/a35abb97008d711c29e5092c9b99117529c02037/client/paper/entry.js#L125

If this is called twice the second time null is returned instead of a reference to the canvas. I quick fix would be to generate a unique id for each whisker canvas.

**Comment by janpaul123** [comment]
Those changes roughly make sense to me! Quick thoughts:
1. What would happen to a blue whisker once it connects? (current whiskers have two colours, red and green; which one would the new `color` attribute replace)
2. Maybe instead of `whisker.onPaperPlaced` use `whisker.on('paperPlaced')` so you can make `whisker` an [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget)?

**Comment by paulsonnentag** [comment]
1. My idea was to use the "moving dot" animation as a state indicator:
    - If the paper is not connected just a static line is projected
    -  If the paper is connected the dot starts moving
2. I agree, makes more sense to use EventTarget here

**Comment by janpaul123** [comment]
Ah that's a good idea! Maybe we should drop the two different colours in the default case too and just use the moving dot animation? In any case, sounds good to me. 💯
